test/acceptance/tests/basic/basic_test.go

```diff
@@ -128,6 +128,31 @@ func TestBasic(t *testing.T) {
 			require.NoError(t, json.Unmarshal([]byte(taskListOut), &tasks))
 			require.Len(t, tasks.TaskARNs, 1)
 
+			// Create an intention.
```

Comment:
Q: This intention is needed because...we specify a default deny when ACLs are enabled?

Reply:
Yep, exactly. When ACL default policy is "default-deny" intentions are also "default-deny" and so we need to explicitly add one. 

It's also mentioned in the [Intentions](https://www.consul.io/docs/connect/intentions#intentions) docs but it's a bit non-obvious:

> The default intention behavior is defined by the default ACL policy. If the default ACL policy is "allow all", then all Connect connections are allowed by default. If the default ACL policy is "deny all", then all Connect connections or requests are denied by default.

Reply:
Actually, I lied. I discovered that default-deny needs to be set on both clients and servers for it to be the case, and I was only setting it on servers. I'll update to make clients use default-deny as well

Reply:
I'll update acceptance test as well to check that we can't connect unless we add an intention.